cli/cmd/recover.go

```diff
@@ -0,0 +1,218 @@
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package cmd
+
+import (
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/hex"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/edgelesssys/contrast/internal/atls"
+	"github.com/edgelesssys/contrast/internal/attestation/snp"
+	"github.com/edgelesssys/contrast/internal/fsstore"
+	"github.com/edgelesssys/contrast/internal/grpc/dialer"
+	"github.com/edgelesssys/contrast/internal/manifest"
+	"github.com/edgelesssys/contrast/internal/recoveryapi"
+	"github.com/edgelesssys/contrast/internal/userapi"
+	"github.com/spf13/cobra"
+)
+
+// NewRecoverCmd creates the contrast recover subcommand.
+func NewRecoverCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "recover [flags]",
+		Short: "recover a contrast deployment after restart",
+		Long: `Recover a contrast deployment after restart.
+
+The state of the Coordinator is stored protected on a persistent volume.
+After a restart, the Coordinator requires the seed to derive the signing
+key and verify the state integrity.
+
+The recover command is used to provide the seed to the Coordinator.`,
+		RunE: withTelemetry(runRecover),
+	}
+
+	cmd.Flags().StringP("manifest", "m", manifestFilename, "path to manifest (.json) file")
+	cmd.Flags().StringP("coordinator", "c", "", "endpoint the coordinator can be reached at")
+	must(cobra.MarkFlagRequired(cmd.Flags(), "coordinator"))
+	cmd.Flags().String("coordinator-policy-hash", DefaultCoordinatorPolicyHash, "override the expected policy hash of the coordinator")
+	cmd.Flags().String("workload-owner-key", workloadOwnerPEM, "path to workload owner key (.pem) file")
+	cmd.Flags().String("seedshare-owner-key", seedshareOwnerPEM, "private key to decrypt the seed share")
+	cmd.Flags().String("seed", seedSharesFilename, "file with the encrypted seed shares")
```

Comment:
Some of these are leftovers, I assume.

Reply:
Why? I think we need all those flags for recovery.

Reply:
Sorry, I forgot that we need the manifest for attestation config.